packages/vite/src/node/optimizer/index.ts

```diff
@@ -1145,14 +1145,28 @@ function isSingleDefaultExport(exports: readonly string[]) {
 }
 
 const lockfileFormats = [
+  // Prefer over package-lock.json as it will only be updated after `npm install` ran
+  {
+    name: '.package-lock.json',
+    path: 'node_modules/.package-lock.json',
+    checkPatches: true,
+    manager: 'npm',
+  },
   { name: 'package-lock.json', checkPatches: true, manager: 'npm' },
   { name: 'yarn.lock', checkPatches: true, manager: 'yarn' }, // Included in lockfile for v2+
+  // Prefer over pnpm-lock.yaml as it will only be updated after `pnpm install` ran
+  {
+    name: 'lock.yaml',
+    path: 'node_modules/.pnpm/lock.yaml',
+    checkPatches: false,
+    manager: 'pnpm',
+  }, // Included in lockfile
   { name: 'pnpm-lock.yaml', checkPatches: false, manager: 'pnpm' }, // Included in lockfile
   { name: 'bun.lockb', checkPatches: true, manager: 'bun' },
```

Comment:
I think we can remove the normal `package-lock.json` and `pnpm-lock.yaml` because we are only interested about  the content in the `node_modules`.

It seems yarn pnp outputs `.yarn/install-state` file and yarn non-pnp outputs `node_modules/.yarn-state.yml`. I guess we can use them for yarn.
For bun, I didn't find any file that contains information about the content of `node_modules`. I think we can leave that one for now.

Reply:
If we want to support yarn 1, we also need to check `node_modules/.yarn-integrity`

Reply:
Thanks for the pointers! I adjusted the paths accordingly and as we now explicitly differentiate between yarn v1 and yarn v2 I have also set the `checkPatches` field to false for v2.